docs/advanced/atomic-swap.md

```diff
@@ -0,0 +1,33 @@
+# Atomic Swaps
+
+---
+
+## Overview
+
+An Atomic Swap is a smart contract technology which makes possible to exchange coins from two different blockchains without having to trust any third party, for example a centralized exchange.
```

Comment:
While interesting to talk about Turing completeness and smart contracts generally, I think this page should focus just on what is necessary to explain atomic swaps. The reader here will likely be fairly educated already on what a smart contract is already. Below, I've taken a stab at tightening up the text.

> An Atomic Swap is a smart contract technology which makes it possible to exchange coins from two different blockchains without having to trust a third party, for example a centralized exchange. 

>Atomic swaps do not require Turning-completeness, but do require both blockchains to have the same hash function and support the ‘OP_CHECKLOCKTIMEVERIFY’ opcode. This makes it possible to create timelocks. 

I am also wondering what is meant by 'the same hash function' here. Decred and Litcoin use different hash functions for mining (SHA-256 vs Blake-256). But I assume you mean a specific type of hash function other than those. As a reader, I'm a little confused. Maybe we can just add a modifier or two even (i.e. "_X type_ hash"), that could clarify things.

Reply:
[The blog post](https://blog.decred.org/2017/09/20/On-Chain-Atomic-Swaps/) covers the prerequisites in detail.

The coins on either side of the swap do not have to use the same hash algo for **mining**, they just need the algo available for use in the tx scripting language.